security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -1014,8 +1037,13 @@ func (sc *SecretCache) sendRetriableRequest(ctx context.Context, csrPEM []byte,
 		var httpRespCode int
 		if isCSR {
 			requestErrorString = fmt.Sprintf("%s CSR", logPrefix)
-			certChainPEM, err = sc.fetcher.CaClient.CSRSign(
-				ctx, reqID, csrPEM, exchangedToken, int64(sc.configOptions.SecretTTL.Seconds()))
+			if withToken {
```

Comment:
same here, simpler if you call the function once.

Reply:
rewrite it and refactor it.